sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/export/CollectionHandle.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.internal.export;
+
+import java.util.AbstractSet;
+import java.util.BitSet;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
+import javax.annotation.Nullable;
+
+/**
+ * A handle for a collection-pipeline of metrics.
+ *
+ * <p>This class provides an efficient means of leasing and tracking exporters.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public final class CollectionHandle {
+  /** The index of this handle. */
+  private final int index;
+
+  private CollectionHandle(int index) {
+    this.index = index;
+  }
+
+  /** Construct a new (efficient) mutable set for tracking collection handles. */
```

Comment:
@jsuereth I thought you said something about complete and working and optimizing in followups :P Just messing

Reply:
I wrote this >6 months ago.   Haven't actually tested it in the current incarnation we have in this SDK, so we'll need to rework the `*Storage` classes to make sure this is actually used well :)